### DISCUSSION TITLE
how can i enable microphone and webcam in MacOS and Linux ?

### DISCUSSION THREAD

**Discussion by skinkairewalker** [wailsapp/wails#3579] [General]
Hello everyone !
how u doing ?

have a way to enable Microphone & Webcam in MacOS & Linux ??
I tested it on both, and both say it's not supported

**Comment by gj1118** [comment]
@skinkairewalker I am getting the same error. Were you able to figure it out please? 
Thanks

**Comment by FirasSharp** [comment]
Can you share what is it that you are doing exactly? 

@skinkairewalker, @gj1118

**Comment by leaanthony** [comment]
Media device access (microphone, webcam) goes through the browser'''s `getUserMedia` API and is gated by OS-level permissions. Here'''s what you need for each platform:

**macOS**
You need to declare `NSMicrophoneUsageDescription` and/or `NSCameraUsageDescription` in your app'''s `Info.plist`. Without these, macOS will silently deny permission.

Edit `build/darwin/Info.plist` in your project and add the keys inside the `<dict>`:

```xml
<key>NSMicrophoneUsageDescription</key>
<string>This app needs microphone access for voice chat.</string>
<key>NSCameraUsageDescription</key>
<string>This app needs camera access for video.</string>
```

This works for both **v2** and **v3** — the file at `build/darwin/Info.plist` is yours to edit and Wails bundles it directly into the app at build time.

**Linux**
Linux WebKit (WebKitGTK) supports `getUserMedia` but the behaviour depends on the installed WebKit version and whether the GStreamer plugins for camera/audio are present. On many distros you need `gstreamer1.0-plugins-good` (or `-bad`) installed. There'''s no separate permission prompt — it'''s controlled by the system PipeWire/PulseAudio permissions.

**Windows**
WebView2 supports `getUserMedia` natively. If it'''s not working, check that the user has granted camera/microphone permission to your app in **Settings → Privacy → Camera/Microphone**.

If you'''re still stuck after checking the above, share your platform, Wails version, and the exact error (or lack of it) and we can dig further.
